streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -907,21 +913,33 @@ private void resetOffsetsIfNeededAndInitializeMetadata(final java.util.function.
         }
     }
 
-    private void initializeTaskTime(final Map<TopicPartition, OffsetAndMetadata> offsetsAndMetadata) {
+    private void initializeTaskTimeAndProcessorMetadata(final Map<TopicPartition, OffsetAndMetadata> offsetsAndMetadata) {
+        ProcessorMetadata finalProcessMetadata = null;
```

Comment:
It seems to be easier to just do:
```
final ProcessorMetadata finalProcessMetadata = new ProcessorMetadata()`
```

And always call `update()` blindly inside the loop. We also don't need the last check:
```
finalProcessMetadata = finalProcessMetadata == null ? ProcessorMetadata.emptyMetadata() : finalProcessMetadata;
```
as we know that it cannot be `null`?

Reply:
Good point. We can do that.